mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -260,13 +326,65 @@ def read_predictions(
         start: datetime,
         end: datetime,
         aggregation_window: typing.Optional[str] = None,
+        limit: typing.Optional[int] = None,
     ) -> typing.Union[
         mm_schemas.ModelEndpointMonitoringMetricValues,
         mm_schemas.ModelEndpointMonitoringMetricNoData,
     ]:
-        raise NotImplementedError
+        if not aggregation_window:
+            logger.warning(
+                "Aggregation window is not provided, defaulting to 10 minute."
+            )
+            aggregation_window = "10m"
+
+        df = self.get_records(
+            table=mm_schemas.TDEngineSuperTables.PREDICTIONS,
+            start=start,
+            end=end,
+            columns=["latency"],
+            filter_query=f"endpoint_id='{endpoint_id}'",
+            agg=["count"],
+            interval=aggregation_window,
+            limit=limit,
+        )
+
+        full_name = mlrun.model_monitoring.helpers.get_invocations_fqn(self.project)
+
+        if df.empty:
+            return mm_schemas.ModelEndpointMonitoringMetricNoData(
+                full_name=full_name,
+                type=mm_schemas.ModelEndpointMonitoringMetricType.METRIC,
+            )
+
+        df["_wend"] = pd.to_datetime(df["_wend"])
+        df.set_index("_wend", inplace=True)
+        return mm_schemas.ModelEndpointMonitoringMetricValues(
+            full_name=full_name,
+            values=list(
+                zip(
+                    df.index,
+                    df["count(latency)"],
+                )
+            ),  # pyright: ignore[reportArgumentType]
+        )
 
     def read_prediction_metric_for_endpoint_if_exists(
         self, endpoint_id: str
     ) -> typing.Optional[mm_schemas.ModelEndpointMonitoringMetric]:
-        raise NotImplementedError
+        # Read just one record, because we just want to check if there is any data for this endpoint_id
```

Comment:
I wonder if we really need to duplicate this method... It's basically the same in both engines.

Reply:
amm it's not 100% similar. In V3IO we need to use strings for the `start` and `end` fields (`0` and `now`) while in TDEngine we need to use datetime format for these params (`datetime.min` and `mlrun.utils.now_date()`)

Reply:
Those two could be extracted to the class though, rather than duplicating the whole thing.